Include/internal/pycore_mmap.h

```diff
@@ -0,0 +1,46 @@
+#ifndef Py_INTERNAL_MMAP_H
+#define Py_INTERNAL_MMAP_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef Py_BUILD_CORE
+#  error "this header requires Py_BUILD_CORE define"
+#endif
+
+#include "pycore_pystate.h"
+
+#if defined(HAVE_PR_SET_VMA_ANON_NAME) && defined(__linux__)
+#  include <linux/prctl.h>
+#  include <sys/prctl.h>
+#endif
+
+#if defined(HAVE_PR_SET_VMA_ANON_NAME) && defined(__linux__) && defined(Py_DEBUG)
+#  define _PyAnnotateMemoryMap(addr, size, name) \
```

Comment:
Do we really need macros? it is just to avoid the `(unsigned long)` cast? If so, I would suggest having a static inline function (but you can have a macro for the no-op case to avoid an empty function) and leave the rest to the compiler. Or if you want to keep a macro, please align the `\` (see PEP-7).